src/components/rule-builder/conditional-logic.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import type { Question, Schema } from '@types';
+import RuleBuilder, { type FormRule } from './rule-builder.component';
+import { v4 as uuidv4 } from 'uuid';
+
+interface ConditionalLogicProps {
+  pageIndex: number;
+  sectionIndex: number;
+  questionIndex: number;
+  question: Question;
+  schema: Schema;
+  rules: Array<FormRule>;
+  handleAddLogic: (fieldId: string) => void;
+  onSchemaChange: (schema: Schema) => void;
+}
+
+const ConditionalLogic: React.FC<ConditionalLogicProps> = ({
+  question,
+  pageIndex,
+  sectionIndex,
+  questionIndex,
+  rules,
+  schema,
+  onSchemaChange,
+  handleAddLogic,
+}) => {
+  const rulesForQuestionIndex = rules?.findIndex((rule: FormRule) => rule?.question === question?.id);
+  const rulesForQuestion =
+    rulesForQuestionIndex !== -1 && rulesForQuestionIndex !== undefined
+      ? rules?.filter((rule: FormRule) => rule?.question === question?.id)
+      : [{ id: uuidv4(), isNewRule: false, question: question?.id }];
+  return (
+    <>
+      {rulesForQuestion.map((rule: FormRule) => (
+        <RuleBuilder
+          ruleId={rule.id}
+          key={question.id}
```

Comment:
Isn't `question.id` basically guaranteed to be non-unique? I.e., every value of `rulesForQuestion` should have the same `question.id`, right?

Reply:
Yeah, looks like it doesn't make sense for the key to be the `question.id` in that case, does it?

Reply:
Yeah. I think it's probably fine to use `rule.id` though? That should be unique-per-tree.